server/metrics/metrics.go

```diff
@@ -208,6 +213,20 @@ func New(config *libconfig.MetricsConfig) *Metrics {
 			},
 			[]string{"name"},
 		),
+		metricDefunctProcessesTotal: prometheus.NewGaugeFunc(
+			prometheus.GaugeOpts{
+				Subsystem: subsystem,
+				Name:      CRIODefunctProcessesTotalKey,
+				Help:      "Total number of defunct processes in the node",
+			},
+			func() float64 {
+				total, err := process.DefunctProcesses()
+				if err == nil {
+					return float64(total)
+				}
+				return 0
```

Comment:
Not sure how to handle the error from `DefunctProcesses()` if it is not `nil`.

Reply:
This looks generally good. We could log the error in case we cannot retrieve them.

Reply:
You could also implement a custom collector to cache the last know value + expose another metric `crio_processes_defunct_timestamp` reporting the "freshness" of the metric.

Reply:
but the way it's implemented right now might be good enough :)